scripts/mk_util.py

```diff
@@ -2358,6 +2365,8 @@ def mk_config():
             CPPFLAGS     = '%s -DNDEBUG -D_EXTERNAL_RELEASE' % CPPFLAGS
         if TRACE or DEBUG_MODE:
             CPPFLAGS     = '%s -D_TRACE' % CPPFLAGS
+        # '$$' is needed to escape '$' in the makefile, the result should be '$ORIGIN'
+        JAVA_EXTERNAL_LIBRARY_LOADING_FLAG = '-Wl,-rpath,\'$$ORIGIN\' ' if use_java_external_library_loading() else ''
```

Comment:
@kfriedberger I seem to remember that using ``rpath``  is considered insecure if the library is installed (https://bugs.archlinux.org/task/14049). Is this necessary? 

Note for others looking at this: for Linux it looks like ``'$ORIGIN'`` has a special meaning which is the directory of the library. So the system linker will look for ``libz3.so`` in the same directory as the ``libz3java.so``